### PR TITLE
CR-1194 NISRA CE4 code

### PR DESCRIPTION
**Pull request by Paul-Joel**
Signed-off-by: Paul-Joel <paul.joel@ons.gov.uk>

# Motivation and Context
NI CE Manager UAC's start 'CE4' but are not dealt with by us. Inform the user

# What has changed
Code checks submitted UAC to see if it starts 'CE4'. If 'CE4' and on /ni user gets new error page, directing them to NISRA Communal Portal. If /en or /cy, user gets error for 'code for Northern Ireland'
Added unit tests to check, and updated incorrect existing test for missing uac

No Cucumber updates required

# How to test?
Enter a UAC beginning CE4 on all 3 start pages, and check appropriate error is shown

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1194

### CODE REVIEW

File: app/start_handlers.py [hunk 1]
```diff
@@ -222,6 +231,28 @@ async def get(self, request):
...
+@start_routes.view('/ni/start/code-for-ce-manager/')
+class StartNICE4Code(StartCommon):
+    @aiohttp_jinja2.template('start-code-for-ni-ce-manager.html')
+    async def get(self, request):
+        self.setup_request(request)
+        display_region = 'ni'
+        self.log_entry(request, display_region + '/start/code-for-ce-manager')
+
+        locale = 'en'
+        page_title = 'This access code is not part of the census for England and Wales'
```
Comment (cookel2) [inline app/start_handlers.py:243]: Does this need to be translated into Welsh when it get displayed in http://localhost:9092/cy/start/code-for-northern-ireland/?

**Review by MartinHumphrey — CHANGES_REQUESTED** [review]
Minor changes.

File: app/start_handlers.py [hunk 2]
```diff
@@ -222,6 +231,28 @@ async def get(self, request):
...
+
+        await forget(request)
+
+        return {
+            'display_region': display_region,
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/start_handlers.py]: Yep, removed

File: app/start_handlers.py [hunk 3]
```diff
@@ -222,6 +231,28 @@ async def get(self, request):
...
+            'locale': locale,
+            'page_title': page_title,
+            'page_url': View.gen_page_url(request),
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/start_handlers.py]: Yep, removed

File: app/start_handlers.py [hunk 4]
```diff
@@ -222,6 +231,28 @@ async def get(self, request):
...
+    @aiohttp_jinja2.template('start-code-for-ni-ce-manager.html')
+    async def get(self, request):
+        self.setup_request(request)
+        display_region = 'ni'
+        self.log_entry(request, display_region + '/start/code-for-ce-manager')
+
+        locale = 'en'
+        page_title = 'This access code is not part of the census for England and Wales'
+
+        await forget(request)
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/start_handlers.py]: Agreed - removed

File: app/templates/start-code-for-ni-ce-manager.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{% extends 'base-ni.html' %}
+
+{% block main %}
+    <h1>You need to visit the Communal Establishment Portal</h1>
+
+    <p>Enter your manager access code on the <a href="https://nisra.gov.uk/census-ce/">Communal Establishment Portal website</a> to start your census.</p>
+
+    <p>Alternatively, you can <a href="{{ contact_us_link }}">contact us</a> for help.</p>
+
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/templates/start-code-for-ni-ce-manager.html]: Fixed

File: app/start_handlers.py (hunk 1 above)
Comment (Paul-Joel) [reply app/start_handlers.py:247]: As discussed, this can only be for NI, so is only in english. The route for en/cy does include translation markers.

**Review by MartinHumphrey — APPROVED** [review]
Changes are OK. But not sure about merge of new Pipfile from master?

**Review by cookel2 — APPROVED** [review]
all tested and looks correct to me

**Pull request merged**